fastly/resource_fastly_integration.go

```diff
@@ -0,0 +1,149 @@
+package fastly
+
+import (
+	"context"
+	"log"
+
+	gofastly "github.com/fastly/go-fastly/v9/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceFastlyIntegration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceFastlyIntegrationCreate,
+		ReadContext:   resourceFastlyIntegrationRead,
+		UpdateContext: resourceFastlyIntegrationUpdate,
+		DeleteContext: resourceFastlyIntegrationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"config": {
+				Type:        schema.TypeMap,
+				Required:    true,
+				Description: "Configuration specific to the integration `type`.",
+				Elem:        schema.TypeString,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User submitted description of the integration.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "User submitted name of the integration.",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Type of the integration. One of: `mailinglist`, `microsoftteams`, `newrelic`, `pagerduty`, `slack`, `webhook`.",
```

Comment:
The following suggested edit adds validation for the allowed inputs (it will also require the following import):

```
"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
```

```suggestion
				Description: "Type of the integration. One of: `mailinglist`, `microsoftteams`, `newrelic`, `pagerduty`, `slack`, `webhook`.",
				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(
					[]string{"mailinglist", "microsoftteams", "newrelic", "pagerduty", "slack", "webhook"},
					false,
				)),
```